src/commissioner/ExampleCommissioningStateMachine.cpp

```diff
@@ -0,0 +1,306 @@
+    {
+        return mFactory.CreateInvokingAddTrustedRootCertificate(event.Get<OperationalCredentials>());
+    }
+    else if (state.Is<InvokingAddTrustedRootCertificate>() && event.Is<OperationalCredentials>())
+    {
+        return mFactory.CreateInvokingAddNOC(event.Get<OperationalCredentials>());
+    }
+    else if (state.Is<InvokingAddNOC>() && event.Is<OperationalCredentials>())
+    {
+        return mFactory.CreateOpCredsWritten();
+    }
+    else if (state.Is<OpCredsWritten>() && event.Is<InitiateNetworkConfiguration>())
+    {
+        return mFactory.CreateReadingNetworkFeatureMap();
+    }
+    else if (state.Is<OpCredsWritten>() && event.Is<SkipNetworkConfiguration>())
+    {
+        return mFactory.CreateNetworkEnabled();
+    }
+    else if (state.Is<ReadingNetworkFeatureMap>() && event.Is<NetworkFeatureMap>())
+    {
+        return mFactory.CreateNetworkFeatureMapRead(event.Get<NetworkFeatureMap>());
+    }
+    else if (state.Is<NetworkFeatureMapRead>() && event.Is<AddOrUpdateWiFiNetwork>())
+    {
+        return mFactory.CreateInvokingAddOrUpdateWiFiNetwork(event.Get<AddOrUpdateWiFiNetwork>());
+    }
+    else if (state.Is<NetworkFeatureMapRead>() && event.Is<AddOrUpdateThreadNetwork>())
+    {
+        return mFactory.CreateInvokingAddOrUpdateThreadNetwork(event.Get<AddOrUpdateThreadNetwork>());
+    }
+    else if ((state.Is<InvokingAddOrUpdateWiFiNetwork>() || state.Is<InvokingAddOrUpdateThreadNetwork>()) && event.Is<NetworkId>())
+    {
+        return mFactory.CreateNetworkAdded(event.Get<NetworkId>());
+    }
+    else if (state.Is<NetworkAdded>() && event.Is<NetworkId>())
+    {
+        return mFactory.CreateInvokingConnectNetwork(event.Get<NetworkId>());
+    }
+    else if (state.Is<InvokingConnectNetwork>() && event.Is<Success>())
+    {
+        return mFactory.CreateNetworkEnabled();
+    }
+    else if (state.Is<NetworkEnabled>() && event.Is<InitiateOperationalDiscovery>())
+    {
+        return mFactory.CreateOperationalDiscovery();
+    }
+    else if (state.Is<OperationalDiscovery>() && event.Is<OperationalRecord>())
+    {
+        return mFactory.CreateInitiatingCase(event.Get<OperationalRecord>());
+    }
+    else if (state.Is<InitiatingCase>() && event.Is<Success>())
+    {
+        return mFactory.CreateCaseComplete();
+    }
+    else if (state.Is<CaseComplete>() && event.Is<InvokeCommissioningComplete>())
+    {
+        return mFactory.CreateInvokingCommissioningComplete();
+    }
+    else if (state.Is<InvokingCommissioningComplete>() && event.Is<Success>())
+    {
+        return mFactory.CreateCommissioningComplete();
+    }
+    else if (event.Is<Failure>())
+    {
+        this->mCtx.CancelTimeout();
+        return mFactory.CreateFailed();
+    }
+    else if (event.Is<Shutdown>())
+    {
+        this->mCtx.CancelTimeout();
+        return mFactory.CreateIdle();
+    }
+    else
+    {
+        return {};
```

Comment:
It might be worth logging the state and the event if reaching the `else` branch.